ComfyUI API Key Integration (#87)
Comfy-Org/docs: essentials/comfyui-server/api-key-integration.mdx

---
title: 'Integration of API Key to use ComfyUI API nodes'
description: 'This article explains how to use ComfyUI API Key to call ComfyUI API nodes in headless mode'
sidebarTitle: 'API Key Integration'
---

> Starting from [PR #8041](https://github.com/comfyanonymous/ComfyUI/pull/8041), ComfyUI supports directly using built-in API nodes through API Keys, without requiring a specific frontend interface (you can even run completely without a frontend).

Reviewer: ```suggestion
Starting from [PR #8041](https://github.com/comfyanonymous/ComfyUI/pull/8041), ComfyUI supports directly using built-in API nodes through API Keys, without requiring a specific frontend interface (you can even run without a frontend).
```

This means you can create workflows that combine:
- local OS models
- tools from the custom node community
- popular paid models

> run everything together via the local Comfy webserver API, letting it handle all the orchestration.

Reviewer: ```suggestion
Then run everything together by simply sending the prompt to the Comfy webserver API, letting it handle all the orchestration.
```

> This is helpful for users who want to use Comfy as a backend service, run Comfy via command line, have their own frontend, etc.

Reviewer: ```suggestion
This is helpful for users who want to use Comfy as a backend service, via the command line, with their own frontend, etc.
```

      "prompt_upsampling": false,
      "seed": 589991183902375,
      "aspect_ratio": "1:1",
      "raw": false,
      "image_prompt_strength": 0.4000000000000001,
      "image_prompt": [
        "14",
        0
      ]
    },
    "class_type": "FluxProUltraImageNode",
    "_meta": {
      "title": "Flux 1.1 [pro] Ultra Image"
    }
  },
  "12": {
    "inputs": {
      "filename_prefix": "ComfyUI",
      "images": [
        "11",
        0
      ]
    },
    "class_type": "SaveImage",
    "_meta": {
      "title": "Save Image"
    }
  },
  "14": {
    "inputs": {
      "image": "example.png"
    },
    "class_type": "LoadImage",
    "_meta": {
      "title": "Load Image"
    }
  }
}"""

prompt = json.loads(workflow_with_api_nodes)
payload = {
    "prompt": prompt,
    # Add the `api_key_comfy_org` to the payload.
    # You can first get the key from the associated user if handling multiple clients.
    "extra_data": {
        "api_key_comfy_org": "comfyui-87d01e28d*******************************************************"  # replace with actual key
    },
}
data = json.dumps(payload).encode("utf-8")
req = request.Request(f"{SERVER_URL}/prompt", data=data)

> 
> # Send the request

Reviewer: ```suggestion
```

We can remove this part of the example script.
